WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -518,7 +518,13 @@ private extension CardReaderConnectionController {
            underlyingError.isSoftwareUpdateError {
             return onUpdateFailed(error: error)
         }
-        showConnectionFailed(error: error)
+
+        if case CardReaderServiceError.connection(underlyingError: let underlyingError) = error, underlyingError.isLocationIDMissingError {
+            return onConnectionFailed(error: error)
```

Comment:
I used that name first too, and didn't like it because it's so similar to the enclosing method name. 

Did you want to inline the error creation here for a particular reason? I quite liked extracting it to another method which then differentiated between the default alert and the missing address alert. I'm going to go back to that for now, and try to name stuff better, as the previous code was handling both alert cases.

Reply:
I have no preference here, to be honest. This looked slightly less verbose to me, and that's why I went with it 😄